src/drivers/intel/cavs/dmic.c

```diff
@@ -134,13 +133,24 @@ struct pdm_controllers_configuration {
 	(Q_SHIFT_BITS_32(30, DB2LIN_FIXED_OUTPUT_QY, 30))
 
 /* tracing */
-#define trace_dmic(__e, ...) \
```

Comment:
Why did you cut it to just `d`? `dmic` felt ok and `d` can be interpreted as `debug`.

Reply:
trace_ macros names are too long (esp. the version with ids) imo, and they are sometimes followed by very long messages which creates too long lines. I'd keep them short as long as they do not collide (preprocessor will let you know). Btw it could default to trace() as well if trace class is redef-ed appropriately locally.

Reply:
Would a scheme such as trace_t (for trace_event), trace_e (for trace_error) and trace_v (for trace_verbose), and trace_it/trace_ie/trace_iv (for the counterparts with IDs) be good? This would also not include the component name where it isn't needed but breaks down if you ever need to use these traces outside the implementation C file.

Reply:
I think our end goal should be the API described by @paulstelian97, where we can also pass in our object handle i.e.
```trace_err(object, "blah blah");```
The object would contain all the necessary ID, control data so that we could turn trace ON/OFF from userspace.

Reply:
@lgirdwood @paulstelian97 Indeed the end goal would be to have something similar with dev_dbg/dev_err/dev_info etc in the Linux kernel.

Where the object passed as the first argument will keep the whole information needed.

Reply:
Added in the goals mentioned in #2172

Reply:
We can try do to what you suggest, but for this PR what we have here is enough. It's more of a cleanup that will also make refactoring that u plan easier